### PR TITLE
[DR-1574] Add AppSec GitHub Trivy Action

### PR DESCRIPTION
**Pull request by dinvlad**
This change adds Trivy security scan for the Docker image, as discussed in the Workbench Cross-Team meeting.

The action will run on every PR and give you feedback if the Docker image contains critical OS vulnerabilities.

More info:
https://github.com/broadinstitute/dsp-appsec-trivy-action
https://github.com/broadinstitute/dsp-appsec-blessed-images

### CODE REVIEW

**Comment by smark88** [comment]
why make extra images vs just scanning what already is built ?

**Comment by dinvlad** [comment]
@smark88 we rebuild them here to provide feedback for the PR. It would be hard to sequence "scan after image build" otherwise, unless you integrate it into your regular CI pipeline.

However, our intention is to keep it as a separate GH Action both for consistency with other repos, and because it doesn't have to interrupt your builds - you just need to consult the results before merging. If it finds Criticals and you decide to ignore them, then the reason should be documented, per our compliance reqs.

Does that make sense?

**Comment by smark88** [comment]
> @smark88 we rebuild them here to provide feedback for the PR. It would be hard to sequence "scan after image build" otherwise, unless you integrate it into your regular CI pipeline.
> 
> However, our intention is to keep it as a separate GH Action both for consistency with other repos, and because it doesn't have to interrupt your builds - you just need to consult the results before merging. If it finds Criticals and you decide to ignore them, then the reason should be documented, per our compliance reqs.
> 
> Does that make sense?

lgtm

**Review by smark88 — APPROVED** [review]

**Review by fboulnois — COMMENTED** [review]
Should we integrate this in one of our other actions that runs on PR?

**Comment by nmalfroy** [comment]
I wonder if having it set to a separate action gives us the option of having seperate "passes security" scan badge?

**Comment by pshapiro4broad** [comment]
I like the idea of having different workflows for different operations, which would make it easier to "share" workflow code across repos (and keep the size of our workflows more manageable). But I don't like the idea of having to look in many places to see what workflows are run on PRs. Would it be possible to trigger this workflow from our current PR workflow?

**Comment by dinvlad** [comment]
Great points here! The main reason why we split it into a separate Action is so it doesn't "break" your builds when it finds issues (which won't happen often, though). This was a sticking point with other teams, who wanted to be able to deploy to prod even despite the errors. Whenever that happens, they should document why a finding was ignored with AppSec, per our compliance requirements.

So you're free to choose whether to keep it separate or to integrate as part of another Action. We would prefer to keep it within Actions however (as opposed to another CI), since we're trying to harmonize how we do this across many teams 😃 

Also, I don't think one GitHub Workflow is able to trigger another, at least not easily.

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    # Parse Dockerfile and build, scan image if a "blessed" base image is not used
+    name: DSP AppSec Trivy check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      # fetch JDK
+      - uses: actions/setup-java@v1
+        with:
+          java-version: '11'
```
Comment (nmalfroy) [inline .github/workflows/trivy.yml]: FWIW, we're currently using 8 to build and run.  Don't know if that'll be an issue here

**Review by fboulnois — APPROVED** [review]
lgtm

**Pull request merged**